setup.py

```diff
@@ -1,18 +1,23 @@
 import setuptools
 import os
 
+
 def read(fname):
+    """
+    Read contents of file as a string.
+    """
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
+
 base_packages = [
-    "scikit-learn>=0.22.2",
```

Comment:
I am thinking if imposing such strict dependencies is a good choice.
If we think about projects that e.g. want to use older versions of pandas or numpy,  then we impose very strict requirements here. What do you think?

Reply:
I am thinking about this now, and even with SHAP dependency, we could just stick to the current version.
It has the new Explainer API, and whichever changes they make,  it should not affect it much. Maybe a good idea is to make a PR, and then just run the tests, but then we can close the PR. What do you think?

Reply:
I'm also having second thoughts on this approach. You get into this complicated web of dependencies where not all packages work together with the latest versions. Best to let `pip` manage that kind of stuff instead of manually fixing the latest version.

In the process though, I made some other improvements to the code, which I will leave.